### PR TITLE
[WALL] Lubega / WALL-4911 / Add wallet utils to deriv-utils library

### PR DESCRIPTION
**Pull request by lubega-deriv**
Added the following wallets utils into formatUtils:
- [x] getFormattedDateString
- [x] getFormattedTimeString
- [x] getAdjustedDate
- [x] parseCryptoLongcode

### CODE REVIEW

**Review by heorhi-deriv — APPROVED** [review]

**Review by jim-deriv — APPROVED** [review]

File: src/utils/format.utils.ts [hunk 1]
```diff
@@ -47,3 +47,165 @@ export const formatMoney = (number: number, options?: FormatMoneyOptions) => {
...
+ * // => "2023-05-15"
+ *
+ * @example
+ * // Returns date in 'DD MMM YYYY' format
+ * getFormattedDateString('2023-05-15', {}, 'DD MMM YYYY');
+ * // => "15 May 2023"
+ */
+export const getFormattedDateString = (
+    dateInput: Date | number | string,
+    options: Intl.DateTimeFormatOptions = { day: "2-digit", month: "2-digit", year: "numeric" },
```
Comment (niloofar-deriv) [inline src/utils/format.utils.ts:76]: [SUGGESTION] Can we encapsulate `options`, `format`, `unix` into a single object, similar to how we handle `formatMoney` util? This approach would simplify the process for users by allowing them to pass only the necessary properties. 🤔

File: src/utils/format.utils.ts [hunk 2]
```diff
@@ -47,3 +47,165 @@ export const formatMoney = (number: number, options?: FormatMoneyOptions) => {
...
+    format = "YYYY-MM-DD",
+    unix = false,
+): string => {
+    let dateObj: Date;
+    const dateOptions: Intl.DateTimeFormatOptions = { ...options };
+
+    if (typeof dateInput === "number" && unix) {
+        dateObj = new Date(dateInput * 1000);
+    } else if (typeof dateInput === "string" || dateInput instanceof Date) {
+        dateObj = new Date(dateInput);
```
Comment (shayan-deriv) [inline src/utils/format.utils.ts:86]:
```suggestion
        dateObj = new Date(dateInput);
        if (isNaN(dateObj.getTime())) {
            throw new Error("Invalid date input");
        }
```

example: the dateInput can be `'11:26:33'` which can't be passed to `new Date()`

File: src/utils/format.utils.ts [hunk 3]
```diff
@@ -47,3 +47,165 @@ export const formatMoney = (number: number, options?: FormatMoneyOptions) => {
...
+ * getFormattedTimeString(1684159800, true);
+ * // => "14:30:00 GMT"
+ */
+export const getFormattedTimeString = (dateInput: Date | number | string, unix = false): string => {
+    let dateObj: Date;
+
+    if (typeof dateInput === "number" && unix) {
+        dateObj = new Date(dateInput * 1000);
+    } else if (typeof dateInput === "string" || dateInput instanceof Date) {
+        dateObj = new Date(dateInput);
```
Comment (shayan-deriv) [inline src/utils/format.utils.ts:141]: same here . please add `isNan` check

File: src/utils/format.utils.ts [hunk 4]
```diff
@@ -47,3 +47,165 @@ export const formatMoney = (number: number, options?: FormatMoneyOptions) => {
...
+ *
+ * @example
+ * // Returns a Date object 10 days in the past
+ * getAdjustedDate(10, "days", "subtract");
+ */
+export const getAdjustedDate = (
+    amount: number,
+    type: "days" | "years" = "days",
+    operation: "add" | "subtract" = "add",
+): Date => {
```
Comment (shayan-deriv) [inline src/utils/format.utils.ts:179]:
```suggestion
): Date => {
if (amount < 0) throw new Error("Amount must be a positive number.");
```

File: src/utils/format.utils.ts [hunk 5]
```diff
@@ -47,3 +47,165 @@ export const formatMoney = (number: number, options?: FormatMoneyOptions) => {
...
+/**
+ * Parses a cryptocurrency longcode string to extract address hash and blockchain hash.
+ *
+ * @param {string} longcode - The cryptocurrency longcode string to parse.
+ *
+ * @returns {Object} An object containing:
+ *   - addressHash: The extracted address hash.
+ *   - blockchainHash: The extracted blockchain hash.
+ *   - splitLongcode: An array of the longcode split by commas.
+ *
```
Comment (shayan-deriv) [inline src/utils/format.utils.ts:201]: can we add an example ?

**Review by shayan-deriv — APPROVED** [review]

**Review by niloofar-deriv — APPROVED** [review]

**Pull request merged**